nest/src/token/pUSD.sol

```diff
@@ -6,137 +6,493 @@ import { Initializable } from "@openzeppelin/contracts-upgradeable/proxy/utils/I
+     */
+    function _authorizeUpgrade(
+        address newImplementation
+    ) internal override(ComponentToken, UUPSUpgradeable) onlyRole(UPGRADER_ROLE) {
+        super._authorizeUpgrade(newImplementation); // Call ComponentToken's checks
     }
 
-    function symbol() public pure override returns (string memory) {
-        return "pUSD";
+    // ========== VIEW FUNCTIONS ==========
+
+    /**
+     * @notice Get the current vault address
+     * @return Address of the current vault
+     */
+    function getVault() external view returns (address) {
+        return address(_getpUSDStorage().boringVault.vault);
     }
 
-    // ========== ADMIN FUNCTIONS ==========
-    function setVault(
-        address newVault
-    ) external onlyRole(VAULT_ADMIN_ROLE) {
-        address oldVault = address(vault);
-        vault = IVault(newVault);
-        emit VaultChanged(oldVault, newVault);
+    /**
+     * @notice Get the current teller address
+     * @return Address of the current teller
+     */
+    function getTeller() external view returns (address) {
+        return address(_getpUSDStorage().boringVault.teller);
     }
 
-    function pause() external onlyRole(PAUSER_ROLE) {
-        paused = true;
-        emit Paused(msg.sender);
+    /**
+     * @notice Get the current AtomicQueue address
+     * @return Address of the current AtomicQueue
+     */
+    function getAtomicQueue() external view returns (address) {
+        return address(_getpUSDStorage().boringVault.atomicQueue);
     }
 
-    function unpause() external onlyRole(PAUSER_ROLE) {
-        paused = false;
-        emit Unpaused(msg.sender);
+    /**
+     * @notice Get the current pUSD version
+     * @return uint256 version of the pUSD contract
+     */
+    function version() public view returns (uint256) {
+        return _getpUSDStorage().version;
     }
 
-    // Required override for UUPSUpgradeable
-    function _authorizeUpgrade(
-        address newImplementation
-    ) internal override onlyRole(UPGRADER_ROLE) { }
+    // ========== COMPONENT TOKEN INTEGRATION ==========
 
-    // ========== ERC20 OVERRIDES ==========
-    function transfer(address to, uint256 amount) public override whenNotPaused returns (bool) {
-        return vault.transferFrom(msg.sender, to, amount);
+    /**
+     * @notice Deposit assets and mint corresponding shares
+     * @param assets Amount of assets to deposit
+     * @param receiver Address that will receive the shares
+     * @param controller Address that will control the shares (unused in this implementation)
+     * @return shares Amount of shares minted
+     */
+    function deposit(
+        uint256 assets,
+        address receiver,
+        address controller,
+        uint256 minimumMint
+    ) public virtual nonReentrant returns (uint256 shares) {
+        if (receiver == address(0)) {
+            revert InvalidReceiver();
+        }
+
+        ITeller teller = ITeller(address(_getpUSDStorage().boringVault.teller));
```

Comment:
`ITeller teller = _getpUSDStorage().boringVault.teller` because it's already `ITeller`